src/main/java/it/gov/pagopa/admissibility/service/commands/operations/DeleteInitiativeServiceImpl.java

```diff
@@ -55,9 +68,12 @@ private Mono<Void> deleteInitiativeCounters(String initiativeId) {
     }
 
     private Mono<Void> deleteOnboardingFamilies(String initiativeId) {
-        return onboardingFamiliesRepository.deleteByInitiativeId(initiativeId)
+
+        return onboardingFamiliesRepository.findByInitiativeIdWithBatch(initiativeId, pageSize)
+                .flatMap(of -> onboardingFamiliesRepository.deleteById(of.getId())
+                        .then(Mono.just(of).delayElement(Duration.ofMillis(delay))), pageSize)
```

Comment:
it should work also without the "of"
MoreOver, you could create a constant
`private static final Mono<?> MONO_DELAY_DELETE = Mono.delayElement(Duration.ofMillis(delay)));`
in order to avoid to create it each time

```suggestion
                        .then(MONO_DELAY_DELETE, pageSize)
```